### ISSUE TITLE
Unable to Replicate the Double Descent Behavior for SVHN. 

### ISSUE THREAD

**Issue by yguooo** [ZhangXiao96/OptimizationVariance#1]
Hi Xiao, I am currently trying to replicate the effect of double descent for SVHN. Here is the parameter that I use (Basically the default one but with dataset chosen to be the SVHN than CIFAR10)
![img1](https://user-images.githubusercontent.com/10891654/101551406-8da11380-3976-11eb-99cb-e476224dd192.png)
![img2](https://user-images.githubusercontent.com/10891654/101551418-909c0400-3976-11eb-805f-b2bd9c72aa38.png)

Here is the setting that we are using: 
```
data_name = 'svhn' #  args[1]     # 'svhn', 'cifar10', 'cifar100'
model_name = 'resnet18' # args[2]    # 'resnet18', 'resnet34', 'vgg16', 'vgg13', 'vgg11'
noise_split = float(args[1])
opt = 'adam' #args[4]
lr = 3e-4 #  float(args[5])
test_id = 0 # int(args[6])
data_root = '..' # args[7]

train_batch_size = 128 
train_epoch = 250
eval_batch_size = 250
```
The noise level we experimented with is : 0, 0.1, 0.2, 0.3, 0.4

I would definitely appreciate the help for replicating the double descent.

**Comment by yguooo** [comment]
I just realize when I use CIFAR10, I could notice a pretty significant double descent phenomenon on test accuracy, but not test loss. I am wondering if my previous plot is just specific to SVHN? Do you mind adding some explanation on the different behavior?

**Comment by ZhangXiao96** [comment]
Hi, double descent cannot be always observed indeed, and you usually should train the model for a very long time, say, 500\~1000 epochs. You may check the details of epoch-wise double descent in this paper: https://arxiv.org/abs/1912.02292. 
For the part of test error and test loss, this paper may help you: https://openreview.net/forum?id=ZAfeFYKUek5.
Hope this helps~

**Comment by yguooo** [comment]
Thanks! Yep, I have just looked at those papers, and I really enjoy both papers. I noticed a very interesting plot:
![image](https://user-images.githubusercontent.com/10891654/101718175-4d20c300-3a66-11eb-85d8-cab7a0ca3cc5.png)

Does that mean if I am looking for the double descent phenomenon, I should simply look at the test accuracy (1-ZO loss), instead of the test loss (by default set to the cross entropy)?

**Comment by ZhangXiao96** [comment]
> Thanks! Yep, I have just looked at those papers, and I really enjoy both papers. I noticed a very interesting plot:
> ![image](https://user-images.githubusercontent.com/10891654/101718175-4d20c300-3a66-11eb-85d8-cab7a0ca3cc5.png)
> 
> Does that mean if I am looking for the double descent phenomenon, I should simply look at the test accuracy (1-ZO loss), instead of the test loss (by default set to the cross entropy)?

Actually, that depends. In my experiments, epoch-wise double descent is observed on the ZO loss. It seems that some papers (I cannot remember which one...) show that in some situations it can also be observed for other losses, yet I have not checked. However, for classification tasks, test error is our real major concern rather than test loss.

**Comment by yguooo** [comment]
Thanks! That really clarifies my confusion.
